### ISSUE TITLE
jobs: jitter exponential backoff

### ISSUE THREAD

**Issue by ajwerner** [cockroachdb/cockroach#69580]
**Is your feature request related to a problem? Please describe.**

In #66889 we added exponential backoff for jobs upon failure of the coordinator or in the face of retriable errors. This backoff, however, is not jitterred. That means that if a problem is caused by the concurrent execution of jobs leading to node failures, then it's likely that all of these jobs will be resumed at the same time.

**Describe the solution you'd like**
We should jitter the backoff of these jobs. Unfortunately it's not super obvious how to do this. We want the jitter to be deterministic. One approach is to hash the job ID and the backoff number to get a uniform random variable and use that to multiply by some jitter factor. I think that'd be cool and reasonably straightforward. 

**Additional context**

Could solve #69349.

**Comment by HonoreDB** [comment]
Straw man: https://github.com/HonoreDB/cockroach/pull/new/jitter_exponential_backoff

**Comment by ajwerner** [comment]
Yeah that idea looks good.  I'd add another cluster setting for the jitter factor which I'd default to +/-15%. 

maybe something like:
```
jitter*((2 * abs(fnv32(id::string, num_runs::string))/((1<<32)-1)) - 1);
```
